src/hotspot/share/opto/loopnode.hpp

```diff
@@ -1625,11 +1628,41 @@ class PhaseIdealLoop : public PhaseTransform {
   // Split Node 'n' through merge point
   RegionNode* split_thru_region(Node* n, RegionNode* region);
   // Split Node 'n' through merge point if there is enough win.
-  Node *split_thru_phi( Node *n, Node *region, int policy );
+  Node *split_thru_phi(Node *n, Node *region, uint policy);
   // Found an If getting its condition-code input from a Phi in the
   // same block.  Split thru the Region.
   void do_split_if(Node *iff, RegionNode** new_false_region = nullptr, RegionNode** new_true_region = nullptr);
 
+private:
+  class SplitWins {
+  private:
+    uint _total_wins;
+    uint _entry_wins;
+    uint _backedge_wins;
+
+  public:
+    SplitWins() :
+      _total_wins(0),
+      _entry_wins(0),
+      _backedge_wins(0) {};
+
+    void reset() {_total_wins = 0; _entry_wins = 0; _backedge_wins = 0;}
+    void add_win(Node* region, int ctrl_index) {
+      if (region->is_Loop() && ctrl_index == LoopNode::EntryControl) {
+        _entry_wins++;
+      }
+      if (region->is_Loop() && ctrl_index == LoopNode::LoopBackControl) {
+        _backedge_wins++;
```

Comment:
Since the corresponding thing is called `LoopBack` in `LoopNode`, calling this `loop_back_wins` would be a little bit more consistent.